Zend/zend_errors.c

```diff
@@ -0,0 +1,202 @@
+#include "SAPI.h"
+#include "ext/standard/html.h"
+
+/* {{{ zend_error_display_cb */
+ZEND_ERROR_CB_API zend_error_display_cb(ZEND_ERROR_CB_HOOK_ARGS)
+{
+	char *buffer = PG(last_error_message);
+
+	if (PG(display_errors) && ((php_get_module_initialized() && !PG(during_request_startup)) || (PG(display_startup_errors)))) {
+		if (PG(xmlrpc_errors)) {
+			php_printf("<?xml version=\"1.0\"?><methodResponse><fault><value><struct><member><name>faultCode</name><value><int>%pd</int></value></member><member><name>faultString</name><value><string>%s:%s in %s on line %d</string></value></member></struct></value></fault></methodResponse>", PG(xmlrpc_error_number), error_type_str, buffer, error_filename, error_lineno);
+		} else {
+			char *prepend_string = INI_STR("error_prepend_string");
+			char *append_string = INI_STR("error_append_string");
+
+			if (PG(html_errors)) {
+				if (type == E_ERROR || type == E_PARSE) {
+					zend_string *buf = php_escape_html_entities((unsigned char*)buffer, strlen(buffer), 0, ENT_COMPAT, NULL);
+					php_printf("%s<br />\n<b>%s</b>:  %s in <b>%s</b> on line <b>%d</b><br />\n%s", STR_PRINT(prepend_string), error_type_str, buf->val, error_filename, error_lineno, STR_PRINT(append_string));
+					zend_string_free(buf);
+				} else {
+					php_printf("%s<br />\n<b>%s</b>:  %s in <b>%s</b> on line <b>%d</b><br />\n%s", STR_PRINT(prepend_string), error_type_str, buffer, error_filename, error_lineno, STR_PRINT(append_string));
+				}
+			} else {
+				/* Write CLI/CGI errors to stderr if display_errors = "stderr" */
+				if ((!strcmp(sapi_module.name, "cli") || !strcmp(sapi_module.name, "cgi")) &&
+					PG(display_errors) == PHP_DISPLAY_ERRORS_STDERR
+				) {
+#ifdef PHP_WIN32
+					fprintf(stderr, "%s: %s in %s on line %u\n", error_type_str, buffer, error_filename, error_lineno);
+					fflush(stderr);
+#else
+					fprintf(stderr, "%s: %s in %s on line %u\n", error_type_str, buffer, error_filename, error_lineno);
+#endif
+				} else {
+					php_printf("%s\n%s: %s in %s on line %d\n%s", STR_PRINT(prepend_string), error_type_str, buffer, error_filename, error_lineno, STR_PRINT(append_string));
+				}
+			}
+		}
+	}
+#if ZEND_DEBUG
+	if (PG(report_zend_debug)) {
+		zend_bool trigger_break;
+
+		switch (type) {
+			case E_ERROR:
+			case E_CORE_ERROR:
+			case E_COMPILE_ERROR:
+			case E_USER_ERROR:
+				trigger_break=1;
+				break;
+			default:
+				trigger_break=0;
+				break;
+		}
+		zend_output_debug_string(trigger_break, "%s(%d) : %s - %s", error_filename, error_lineno, error_type_str, buffer);
+	}
+#endif
+	return SUCCESS;
+}
+/* }}} */
+
+/* {{{ zend_error_log_cb */
+ZEND_ERROR_CB_API zend_error_log_cb(ZEND_ERROR_CB_HOOK_ARGS)
+{
+	char *buffer = PG(last_error_message);
+
+	if (!php_get_module_initialized() || PG(log_errors)) {
+		char *log_buffer;
+#ifdef PHP_WIN32
+		if (type == E_CORE_ERROR || type == E_CORE_WARNING) {
+			syslog(LOG_ALERT, "PHP %s: %s (%s)", error_type_str, buffer, GetCommandLine());
```

Comment:
This needs win32/syslog.h to be included. Or maybe it'd be better to check what header inclusion did provide it in the old file. For the rest it looks fine.